### PR TITLE
chore(refs T30981) Typescript support demosplan UI

### PR DESCRIPTION
**Pull request by ahmad-demos**
https://yaits.demos-deutschland.de/T30981

- WIP: Add Typescript Config.
- WIP: Add DpRadio and DpButton component as Typescript.

Test:
In Core:
- go to `package.json:44`, then add `"@demos-europe/demosplan-ui": "github:demos-europe/demosplan-ui#200e416bbbb47eda2e9e442ada918d3ad3fc5661",`
- delete `node_modules` folder.
- then `yarn install`.
- in Terminal in phpStorm `cd node_modules/@demos-europe/demosplan-ui` -> `yarn install` -> `yarn build:prod` -> `yarn build-tokens`.
- at the end `dp shell jsdev:project`

:information_source: 
We have to use `.vue` when we import Vue-Components in TS-files.
https://stackoverflow.com/questions/52880039/how-to-remove-vue-extension-from-imports-when-using-typescript-in-vue-js

### CODE REVIEW

File: src/components/DpRadio/DpRadioTs.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
+<template>
```
Comment (spiess-demos) [inline src/components/DpRadio/DpRadioTs.vue:1]: Why do you duplicate the component?

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: src/components/DpRadio/DpRadioTs.vue (hunk 1 above)
Comment (salisdemos) [reply src/components/DpRadio/DpRadioTs.vue:1]: And: Do we want to Postfix the Ts-Components?

Comment (ahmad-demos) [reply src/components/DpRadio/DpRadioTs.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/291/commits/9997286261f720374fb6e9eb0b47b1dff1469140

I initially wanted to make two PR's, one for the UI, and the other for the Core, and after test of component with typescript, can I return name of component.
But also we can test TS component with same name, before Merge.

File: src/@types/vue-shims.d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+declare module "*.vue" {
```
Comment (spiess-demos) [inline src/@types/vue-shims.d.ts:1]:
```suggestion
/**
 * Explain to typescript what Vue components are.
 */
declare module '*.vue' {
    import type { DefineComponent } from 'vue'
    const component: DefineComponent<{}, {}, any>
    export default component
}
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]
When running `yarn install` on this branch, i get a completely changed yarn.lock. Maybe you forgot to commit some changes in the yarn.lock?

File: shims-vue.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply shims-vue.d.ts]: done
I did `yarn build:prod`, there is no error!

[the remark this answers is not in the document]
Comment (ahmad-demos) [reply shims-vue.d.ts]: please Review again!

**Review by salisdemos — APPROVED** [review]
looks reasonable

**Review by spiess-demos — APPROVED** [review]
Lgtm!

File: src/components/DpRadio/DpRadio.vue [hunk 4]
```diff
@@ -26,11 +26,12 @@
...
 
-<script>
-import DpLabel from '../DpLabel/DpLabel'
+<script lang="ts">
+import { defineComponent } from 'vue'
+import DpLabel from '../DpLabel/DpLabel.vue'
 import { prefixClassMixin } from '../../mixins'
 
-export default {
+export default defineComponent({
```
Comment (salisdemos) [inline src/components/DpRadio/DpRadio.vue:34]: please use vue2 syntax here

**Review by salisdemos — CHANGES_REQUESTED** [review]

**Review by spiess-demos — APPROVED** [review]
Lgtm + works!

**Pull request merged**